build-logic/build-init-samples/src/main/kotlin/gradlebuild/generate-samples.gradle.kts

```diff
@@ -43,8 +43,8 @@ multiProjectApplicationSampleLanguages.forEach { language ->
 }
 
 fun setupGeneratorTask(language: Language, kind: String, modularizationOption: ModularizationOption) {
-    val buildInitType = "${language.name}-$kind"
-    val capName = language.name.capitalize()
+    val buildInitType = "${language.getName()}-$kind"
```

Comment:
Why making those changes?

Reply:
In https://github.com/gradle/gradle/commit/110b6323a277177182cb35573f3fbd34bef2927e the Language was changed to enum. And then using `.name` returns enum name and doesn't call `.getName()`.  Due to that task names and samples name have for example "CPP" instead of "Cpp" in name and some things fails.

🤔 I wonder if we should rename `.getName()` to something else to not cause confusion. Not sure to what, we already use display name there. Maybe languageName could work.